### ISSUE TITLE
React Native Switch component how to update state?

### ISSUE THREAD

**Issue by nextriot** [LegendApp/legend-state#49]
I've tried a couple variations and can't figure out how to make the switch component react to the toggle event. I assumed like `input` it would update on its own and update the observable, but it doesn't seem to work that way.

Here's what I've tried:

```
import * as React from 'react'
import { useObservable } from '@legendapp/state/react'
import { Legend } from '@legendapp/state/react-native-components'

export const Component = () => {
  const isSelected = useObservable(true)

  isSelected.onChange(value => {
    console.log(value) // `undefined`
  })

  const handleValueChange = value => {
    // value = `undefined`
    console.log(isSelected.get()) // `undefined`
    isSelected.set(!isSelected.get())
  }

  return (
        <Legend.Switch value$={isSelected} onValueChange$={handleValueChange} style$={Styled.switchStyle} />
  )
}
```

I'm not sure how to update the state, or listen and react to state updates.
I've also tried not setting the value in `handleValueChange` which also doesn't update the state, but when I do update the state in handleValueChange I get the following error:

![Error](https://user-images.githubusercontent.com/711457/197857684-5eee4354-1640-4555-bd82-624f9cb588b2.png)

**Comment by nextriot** [comment]
Managed to get it to work using the native event handler:

```
const handleValueChange = (value: boolean) => {
    state.selected.set(value)
}

onValueChange={handleValueChange}
```

I assume that's how it's intended to work?

**Comment by jmeistrich** [comment]
Ah, sorry about that. There was a mistake in the Switch binding which is fixed now in 0.21.2. So you can use it without the change handler as expected.

```jsx
return (
    <Legend.Switch value$={isSelected} style={Styled.switchStyle} />
)
```

Also a little note on style: you can use the regular `style` prop if you're using a static style object. `style$` lets you pass a function that returns a style object based on observable values.
